crates/sui-rosetta/src/construction.rs

```diff
@@ -65,12 +66,15 @@ pub async fn payloads(
         .try_into_data(metadata)?;
     let intent_msg = IntentMessage::new(Intent::default(), data);
     let intent_msg_bytes = bcs::to_bytes(&intent_msg)?;
+    let mut hasher = DefaultHash::default();
+    hasher.update(intent_msg_bytes);
+    let digest = hasher.finalize();
 
     Ok(ConstructionPayloadsResponse {
-        unsigned_transaction: Hex::from_bytes(&intent_msg_bytes),
+        unsigned_transaction: Hex::from_bytes(&digest.digest),
```

Comment:
ah hadn't notice that we sign in Rosetta as well, what is this key btw @patrickkuo ?

Reply:
qq for @patrickkuo, in rossetta spec, it defines https://www.rosetta-api.org/docs/ConstructionApi.html#constructionpayloads
```
 It returns an unsigned transaction blob and a collection of payloads that must be signed by particular AccountIdentifiers using a certain SignatureType.
```

since now require the signature to be committed over blake2b hash of the intent message, should we change this response to return the hash? (i think the answer is yes?)

```
ConstructionPayloadsResponse {
        unsigned_transaction: Hex::from_bytes(&intent_msg_bytes), // change this to the blake2b hash of the tx
        payloads: vec![SigningPayload {
            account_identifier: address.into(),
            hex_bytes: Hex::encode(bcs::to_bytes(&intent_msg)?), // same here
            signature_type: Some(SignatureType::Ed25519),
        }],
    }
```

i think the answer is yes

Reply:
edit: i changed the behavior of the rosetta client to use sign_secure, that i think it's ok to return the intent msg bytes if thats the only place the signature is produced.

Reply:
got it, the hex_bytes is what the offline keystore takes in to sign, then it has to be blake2b prehashed.